library/Backends/Plonk/IR.md

```diff
@@ -0,0 +1,103 @@
+# PLONK IR
+
+This is mainly a translation of https://hackmd.io/@zacwilliamson/r1z1yZTs8 into Haskell
+
+## Binary operations
+
+| Opcode | Opcode Bytes    | Types | Description     | Notes |
+| ------ | --------------- | ----- | --------------- | ----- |
+| add    | `[a], [b], [o]` |       | o = a + b       |       |
+| sub    | `[a], [b], [o]` |       | o = a - b       |       |
+| mul    | `[a], [b], [o]` |       | o = a * b       |       |
+| div    | `[a], [b], [o]` |       | o = a / b       |       |
+| mod    | `[a], [b], [o]` |       | o = a % b       |       |
+| and    | `[a], [b], [o]` |       | `o = a &amp; b` |       |
```

Comment:
This treats the element as a boolean value, right?

What happens if it is out of range?

Do we have or want opcodes for bitwise `and`, `or`, `xor` ?

Reply:
Let's just note this somewhere, e.g. in the types

Reply:
Bitwise operands seem likely to be expensive, let's avoid them for now